controllers/grafana/grafana_controller.go

```diff
@@ -34,6 +34,15 @@ const DefaultClientTimeoutSeconds = 5
 
 var log = logf.Log.WithName(ControllerName)
 
+// +kubebuilder:rbac:groups=integreatly.org,resources=grafanas,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=integreatly.org,resources=grafanas/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=integreatly.org,resources=grafanas/finalizers,verbs=update
```

Comment:
I have very low knowledge about the finalizers in general. I can try to read up on it a bit more and see if i can give a good answer. But i have been asking myself the same thing.

Reply:
@NissesSenap finalizers are used to prevent deletion of a resource before some kind of cleanup happend. From the operators point of view, if a finalizer is set, it sees a resource with a `deletionTimestamp` that is not deleted. The Operators job now is to do whichever cleanup is required and then remove the finalizer to allow Kubernetes to actually remove the resource.

We're not using them anymore because they can also cause problems, e.g. when an Operator is deleted before it's Operand.

We probably don't need those permissions, i'll try it out.

Reply:
That sounds very reasonable. I got the same understanding when I read up on it yesterday.
I have updated the PR.